sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/DefaultSource.scala

```diff
@@ -63,7 +63,9 @@ class DefaultSource extends FileFormat with DataSourceRegister {
     val header = if (csvOptions.headerFlag) {
       firstRow
     } else {
-      firstRow.zipWithIndex.map { case (value, index) => s"C$index" }
+      // firstRow.zipWithIndex.map { case (value, index) => s"C$index" }
+      new BulkCsvReader(rdd.toLocalIterator, csvOptions, null).next()
+        .zipWithIndex.map { case (value, index) => s"C$index" }
```

Comment:
For the current implementation of CSV datasource, I believe we better use `LineCsvReader` since it will reads a single line (including `\n` character).